Fill openweathermap option dialog with setup values

ReneNulschDE opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
This change makes sure that the options selected in the integration setup are loaded in the options dialog too. fixes: #45517

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests


## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #45517


## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [x] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

ReneNulschDE committed:
Fill option dialogs with setup values

MartinHjelmare commented on homeassistant/components/openweathermap/config_flow.py:
```diff
@@ -108,20 +108,19 @@ def _get_options_schema(self):
+    def _get_config_value(self, key):
+        if self.config_entry.options:
+            return self.config_entry.options[key]
+        return self.config_entry.data[key]
```
Please add a test, something like:
```py
assert result["data_schema"]({}) == {
    CONF_MODE: mode_bla_bla,
    CONF_LANGUAGE: language_bla_bla,
}
```

ReneNulschDE replied:
Is this not tested [here](https://github.com/home-assistant/core/blob/8bacfcec50cf523a788010e519c6f28d7c770e3e/tests/components/openweathermap/test_config_flow.py#L120) already?

MartinHjelmare replied:
Those tests are not testing all the different cases of the default form values.

tkdrob committed:
Clean up kira integration constants (#46390)

kangaroomadman committed:
Allow MQTT template light floating point transition (#46385)

Allow to use floating point values for the transition time of the MQTT
template light.

amelchio committed:
Raise ConditionError for time errors (#46250)

emontnemery committed:
Wait for registries to load at startup (#46265)

* Wait for registries to load at startup

* Don't decorate new functions with @bind_hass

* Fix typing errors in zwave_js

* Load registries in async_test_home_assistant

* Tweak

* Typo

* Tweak

* Explicitly silence mypy errors

* Fix tests

* Fix more tests

* Fix test

* Improve docstring

* Wait for registries to load

tkdrob committed:
Use core constants for nissan_leaf (#46401)

tkdrob committed:
Use core constants for nmap_tracker (#46402)

crazyfx1 committed:
Improve knx fan implementation (#46404)

tkdrob committed:
Use core constants for ohmconnect (#46413)

emontnemery committed:
Migrate mobile_app to RestoreEntity (#46391)

Antoni-Czaplicki committed:
Add generic_thermostat unique ID parameter (#46399)

* Add generic_thermostat unique ID parameter

* Add tests for unique id

* Fix flake8

elupus committed:
Add config flow to philips_js (#45784)

* Add config flow to philips_js

* Adjust name of entry to contain serial

* Use device id in event rather than entity id

* Adjust turn on text

* Deprecate all fields

* Be somewhat more explicit in typing

* Switch to direct coordinator access

* Refactor the pluggable action

* Adjust tests a bit

* Minor adjustment

* More adjustments

* Add missing await in update coordinator

* Be more lenient to lack of system info

* Use constant for trigger type and simplify

* Apply suggestions from code review

Co-authored-by: J. Nick Koston <nick@koston.org>

Co-authored-by: J. Nick Koston <nick@koston.org>

chriss158 committed:
Install libpcap-dev for devcontainer (#46106)

homeassistant committed:
[ci skip] Translation update

elupus committed:
Correct errors found on post merge review in philips_js (#46428)

* Correct missed review changes

* Adjust return value for device trigger

* Drop cannot connect

* Always assume there is a unique id

* No need to yield

* Update homeassistant/components/philips_js/media_player.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Move typing to init

* Adjust typing instead of returning lambda

* Explicity return None

* Coerce into int

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

tkdrob committed:
Use core constants for recollect_waste (#46416)

frenck committed:
Postponed evaluation of annotations in core (#46434)

* Postponed evaluation of annotations in core

* Remove unneeded future

tkdrob committed:
Use core constants for plugwise (#46414)

emontnemery committed:
Improve MQTT timeout print (#46398)

tkdrob committed:
Use core constants for rpi_gpio (#46442)

marcelveldt committed:
Enhance platform discovery for zwave_js (#46355)

kingo55 committed:
Fix cmus remote disconnections (#40284)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

WizBangCrash committed:
Add apple tv remote delay command (#46301)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

cpainchaud committed:
Enable TCP KEEPALIVE to RFLink for dead connection detection (#46438)

RFLink compoment when used over TCP protocol suffers a major issue : it doesn't
know when connection is timeout or lost because there is no keepalive mechanism
so it can stay disconnected forever.
I wrote a small patch for the underlying 'python-rflink' library which will enable
TCP KEEPPAlive. On HASSIO side it will just add an optional argument in yml file
which will propagate to python-rflink caller.

tkdrob committed:
Use core constants for roomba (#46441)

janiversen committed:
Allow Modbus "old" config or discovery_info as configuration (#46445)

emontnemery committed:
Make some Area and EntityRegistry member functions callbacks (#46433)

frenck committed:
Upgrade sentry-sdk to 0.20.1 (#46456)

bieniu committed:
Bump brother library to version 0.2.1 (#46421)

frenck committed:
Postponed evaluation of annotations for integrations (#46455)

bdraco committed:
Fix homekit migration not being awaited (#46460)

tkdrob committed:
Use core constants for somfy (#46466)

tkdrob committed:
Use core constants for rflink (#46440)

homeassistant committed:
[ci skip] Translation update

OnFreund committed:
Gracefully handle missing A/V info in Onkyo integration (#46228)

* Gracefully handle missing A/V info

* Do not attempt to query A/V info if unsupported

* Rename _parse_onkyo_tuple

tkdrob committed:
Use core constants for dynalite (#46044)

tkdrob committed:
Use core constants for homeassistant triggers (#46472)

bieniu committed:
Remove unnecessary gethostbyname() from Shelly integration (#46483)

tkdrob committed:
Use core constants for starline (#46471)

tkdrob committed:
Use core constants for simplepush (#46465)

RobBie1221 committed:
Bump dsmr_parser to 0.28, configure keep_alive_interval (#46464)

frenck committed:
None optional hass typing in FlowHandler (#46462)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

thecode committed:
Remove deprecated Synology integration (#46482)

thecode committed:
Remove deprecated xfinity integration (#46484)

ollo69 committed:
Add Asuswrt Config Flow and Scanner Entities (#46468)

* Add Asuswrt config flow (#43948)

* Add AsusWrt Scanner Entity (#44759)

* Add Scanner Entity

- device tracker entity changed from "DeviceScanner" to "ScannerEntity"
- sensors recoded to use "router" class
- config entry review to allow multiple entity (for future use)

* Force checks

* Removed new option and change sensors

* Update test_sensor.py

* Requested changes

* Removed router unique-id

* Update last_activity attr only when available

* Add Options for AsusWRT Scanner Entity (#44808)

* Add Asuswrt config flow (#43948)

* Add AsusWrt Scanner Entity (#44759)

* Add Scanner Entity

- device tracker entity changed from "DeviceScanner" to "ScannerEntity"
- sensors recoded to use "router" class
- config entry review to allow multiple entity (for future use)

* Force checks

* Removed new option and change sensors

* Update test_sensor.py

* Requested changes

* Removed router unique-id

* Update last_activity attr only when available

* Add Options for Scanner Entity

* Fix isort

* Removed "Track New" option

* Add Options for Scanner Entity

* Fix isort

* Removed "Track New" option

* Add test for all the options in the config flow

bdraco committed:
Update HAP-python to 3.3.0 for homekit (#46497)

Changes: https://github.com/ikalchev/HAP-python/compare/v3.2.0...v3.3.0

Noltari committed:
Add AEMET OpenData integration (#45074)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

frenck committed:
Improve Elgato code quality (#46505)

tkdrob committed:
Use core constants for sma (#46501)

tkdrob committed:
Use core constants for rpi_rf (#46500)

tkdrob committed:
Use core constants for rmvtransport (#46502)

homeassistant committed:
[ci skip] Translation update

bdraco committed:
Update tuya for new fan entity model (#45870)

MatthewFlamm committed:
Fix missing condition in nws (#46513)

tkdrob committed:
Use core constants for temper (#46508)

tkdrob committed:
Use core constants for switcher_kis (#46507)

tkdrob committed:
Use core constants for thethingsnetwork (#46520)

tkdrob committed:
Use core constants for trend (#46521)

Commit:
Add hive hub 360 sensors (#46320)

tkdrob committed:
Clean up template (#46509)

foxel committed:
Add keenetic_ndms2 config flow (#38353)

kpine committed:
Add barrier covers to zwave_js integration (#46379)

allenporter committed:
Improve nest defense against broken event loop on shutdown (#46494)

tkdrob committed:
Cleanup unused loggers (#46510)

palfrey committed:
Don't allow recursive secrets loading (#41812)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

javicalle committed:
Add myself to RFLink codeowners (#46511)

janiversen committed:
Update modbus test harness (#44892)

* Update test harness to allow discovery_info testing.

This is a needed update, before the whole config is converted
to the new structure, because it allows test of both the old
and new configuration style.

The following changes have been made:
- Combine test functions into a single base_test.
- Prepare for new mock by combining the 2 common test functions into one.
- Change mock to be only modbusHub
- Do not replace whole modbus class, but only the class that
  connects to pymodbus (modbusHub). This allows test of modbus
  configurations and not only component configurations, and is needed
  when testing discovery activated platform.
- Add test of climate.
  Warning this is merely test of discovery,
  the real cover tests still needs to be added.
- Add test of cover.
  Warning this is merely test of discovery,
  the real cover tests still needs to be added.

* Update comment for old config

* Do not use hass.data, but instead patch pymodbus library.

* Add test of configuration (old/new way as available).

* Move assert to test function.

Make assert more understandable by removing it from the helper.

add base_config_test (check just calls base_test) to make it clear if
test is wanted or just controlling the config.

* use setup_component to load Modbus since it is an integration.

* Optimized flow in test helper.

amelchio committed:
Do not trigger when template is true at startup (#46423)

bdraco committed:
Add support for pre-filtering events to the event bus (#46371)

tkdrob committed:
Use core constants for vasttrafik (#46539)

bdraco committed:
Fix variable name from script refactoring (#46503)

tkdrob committed:
Use core constants for uvc (#46538)

tkdrob committed:
Use core constants for volvooncall (#46543)

tkdrob committed:
Use core constants for wemo (#46544)

bdraco committed:
Fix typing on tuya fan percentage (#46541)

homeassistant committed:
[ci skip] Translation update

raman325 committed:
Add additional supported feature support to universal media player (#44711)

* add additional supported feature support to universal media player

* add missing services

tkdrob committed:
Use core constants for wemo and whois (#46548)

raman325 committed:
Allow users to set device class for universal media player (#46550)

allenporter committed:
Add additional stream HLS payload tests (#46517)

* Add tests for HLS playlist view details

* Add tests for hls playlist payloads

* Update tests/components/stream/test_hls.py

Co-authored-by: uvjustin <46082645+uvjustin@users.noreply.github.com>

* Update tests/components/stream/test_hls.py

Co-authored-by: uvjustin <46082645+uvjustin@users.noreply.github.com>

* Update tests/components/stream/test_hls.py

Co-authored-by: uvjustin <46082645+uvjustin@users.noreply.github.com>

* Update tests/components/stream/test_hls.py

Co-authored-by: uvjustin <46082645+uvjustin@users.noreply.github.com>

* Update tests/components/stream/test_hls.py

Co-authored-by: uvjustin <46082645+uvjustin@users.noreply.github.com>

* Update tests/components/stream/test_hls.py

Co-authored-by: uvjustin <46082645+uvjustin@users.noreply.github.com>

* Update tests/components/stream/test_hls.py

Co-authored-by: uvjustin <46082645+uvjustin@users.noreply.github.com>

Co-authored-by: uvjustin <46082645+uvjustin@users.noreply.github.com>

emontnemery committed:
Remove @home-assistant/core from MQTT codeowners (#46562)

emontnemery committed:
Bump hatasmota to 0.2.9 (#46561)

spike411 committed:
Convert better from byte value to percentage in futurenow (#45042)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

bdraco committed:
Log ffmpeg errors for homekit cameras (#46545)

uvjustin committed:
Use httpx in generic camera (#46576)

* Use httpx in generic camera

* Remove commented out code

thecode committed:
MQTT cover Bugfixes (#46479)

* MQTT cover Bugfixes

* Remove period

ludeeus committed:
Add websocket handlers to hassio (#46571)

nielstron committed:
Limit fronius error messages on failed connection (#45824)

* Do not print several error messages on failed connection

* Change wrapping of exception, implement available

* Simplify exception flow

* Remove unnecessary init

* Add available property to actual entities

* Rebase and formatting

allenporter committed:
Refactor stream worker responsibilities for segmenting into a separate class (#46563)

* Remove stream_worker dependencies on Stream

Removee stream_worker dependencies on Stream and split out the logic
for writing segments to a stream buffer.

* Stop calling internal stream methods

* Update homeassistant/components/stream/worker.py

Co-authored-by: uvjustin <46082645+uvjustin@users.noreply.github.com>

* Reuse self._outputs when creating new streams

Co-authored-by: uvjustin <46082645+uvjustin@users.noreply.github.com>

ludeeus committed:
Fix issue with timeout and error response (#46584)

starkillerOG committed:
Add config flow to Xiaomi Miio switch (#46179)

fronzbot committed:
Upgrade blinkpy to 0.17.0 (#46581)

bdraco committed:
Use shared clientsession for sense (#46419)

frenck committed:
Upgrade sentry-sdk to 0.20.2 (#46590)

tkdrob committed:
Use core constants for xiaomi_aqara (#46551)

homeassistant committed:
[ci skip] Translation update

ollo69 committed:
Bump tuyaha to 0.0.10 and fix set temperature issues (#45732)

marecabo committed:
Add device_class attribute to ESPHome sensor entities (#46595)

bdraco committed:
Remove support for migrating pre-config entry homekit (#46616)

HomeKit pairings and accessory ids from versions 0.109 and earlier are no longer
migrated on upgrade. Users upgrading directly to 2021.3 from 0.109
and older should upgrade to 2021.2 first if they wish to preserve
HomeKit configuration and avoid re-pairing the bridge.

This change does not affect upgrades from 0.110 and later.

bramkragten committed:
Add target to services.yaml (#46410)

Co-authored-by: Franck Nijhof <git@frenck.dev>

elupus committed:
Use explicit open/close for covers (#46602)

frenck committed:
Remove defunct CoinMarketCap integration (#46615)

dmcc committed:
Bump python-vlc-telnet to 2.0.1 (#46608)

Includes corresponding Home Assistant changes to avoid introducing
regressions.

tkdrob committed:
Use core constants for yeelight (#46552)

starkillerOG committed:
Use update coordinator for Xioami Miio subdevices (#46251)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

MartinHjelmare committed:
Fix vlc_telnet state update (#46628)

* Clean up

* Add debug logs

* Fix info lookup

* Handle more errors

* Guard get length and time

tkdrob committed:
Remove unnecessary constants from universal (#46537)

allenporter committed:
Separate HLS logic out of core StreamOutput to prepare for discontinuity (#46610)

Separate the HLS stream view logic out of StreamOutput since the hls
stream view is about to get more complex to track discontinuities. This
makes the idle timeout, shutdown, and coupling between hls and record
more explicit.

marthoc committed:
Fix climate hold bug in ecobee (#46613)

marecabo committed:
Do not provide icon if device class is set in ESPHome config (#46650)

allenporter committed:
Fix exception in stream idle callback (#46642)

* Fix exception in stream idle callback

Fix bug where idle timeout callback fails if the stream previously exited.

* Add a test for stream idle timeout after stream is stopped.

* Add clarifying comment to idle timer clear method

* Clear hls timer only on stop

esev committed:
Fix KeyError in comfoconnect percentage (#46654)

scop committed:
Fix version of pip in tox (#46656)

homeassistant committed:
[ci skip] Translation update

allenporter committed:
Fix stream keepalive on startup (#46640)

This was accidentally dropped in a previous PR that refactored the interaction
between stream and camera. This is difficult to test from the existing
preload stream tests, so leaving untested for now.

mdz committed:
Add SmartTub integration (#37775)

Co-authored-by: J. Nick Koston <nick@koston.org>

dmcc committed:
Fix media_pause in vlc_telnet (#46675)

The underlying python-telnet-vlc pause() function toggles play/pause, so
we need to check our state before calling it.

balloob committed:
Fix Cloud Google/Alexa check (#46681)

allenporter committed:
Add back block_until_done calls removed in PR 46610 (#46680)

leikoilja committed:
Add sensors to fetch Habitica tasks (#38910)

* Adding sensors to fetch habitica tasks

* PR changes and rebase

* Fixing pylint

* Fixing failed test dependancy

* Generating requirements

* Apply suggestions from code review

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* PR changes

* Update homeassistant/components/habitica/config_flow.py

Thank you, @MartinHjelmare

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* PR Changes

* Fix failing test

* Update tests/components/habitica/test_config_flow.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Fixing linting and imports

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

esev committed:
Remove exception handling for AttributeError in wemo (#46674)

starkillerOG committed:
Fix multiple motion blinds gateways (#46622)

local variable multicast was undefined for a second or more gateway that was setup.

badguy99 committed:
Home connect use consts (#46659)

* Use more consts

* black

* re-black with black, version 20.8b1

allenporter committed:
Fix flaky stream tests due to race in idle timeout callback (#46687)

ollo69 committed:
Fix Tuya Option Flow tests (#46651)

tkdrob committed:
Clean up xbee (#46549)

esev committed:
Increase async_timeout for wemo update polling (#46649)

esev committed:
Remove calls to wemo.reconnect_with_device (#46646)

cgtobi committed:
Bump RMVtransport to v0.3.0 (#46691)

gsemet committed:
Discover HRT4-ZW / SRT321 SetPoint in zwave_js (#46625)

Missing a specific class to allow discovery of the setpoint command.

Fix #46570

Signed-off-by: Gaetan Semet <gaetan@xeberon.net>

bdraco committed:
Update zha to use new fan entity model (#45758)

* Update zha to use new fan entity model

* fixes

* tweaks for zha

* pylint

* augment test cover

esev committed:
Centralize wemo exception handling (#46705)

esev committed:
Perform wemo state update quickly after a timeout (#46702)

* Perform state update quickly after a timeout

* with async_timeout.timeout(...) -> async with async_timeout.timeout(...)

homeassistant committed:
[ci skip] Translation update

uvjustin committed:
Bump pylutron 0.2.7 (#46717)

bdraco committed:
Switch ssdp to be async by using async_upnp_client for scanning (#46554)

SSDP scans no longer runs in the executor

This is an interim step that converts the async_upnp_client
response to netdisco's object to ensure fully backwards
compatibility

frenck committed:
Add selectors to Input * service definitions (#46652)

frenck committed:
Add selectors to Alert service definitions (#46627)

frenck committed:
Add selectors to Camera service definitions (#46630)

frenck committed:
Add selectors to Switch service definitions (#46635)

frenck committed:
Add selectors to Downloader service definitions (#46638)

frenck committed:
Add selectors to Fan service definitions (#46639)

frenck committed:
Add selectors to Cover service definitions (#46634)

Co-authored-by: Tobias Sauerwein <cgtobi@users.noreply.github.com>

frenck committed:
Add selectors to Automation service definitions (#46629)

frenck committed:
Add selectors to Climate service definitions (#46632)

frenck committed:
Add selectors to Alarm Control Panel service definitions (#46626)

frenck committed:
Upgrade and constrain httplib2>=0.19.0 (#46725)

Noltari committed:
ubus: switch to pypi library (#46690)

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

frenck committed:
Add selectors to Counter service definitions (#46633)

frenck committed:
Add advanced service parameter flag (#46727)

allenporter committed:
Add discontinuity support to HLS streams and fix nest expiring stream urls (#46683)

* Support HLS stream discontinuity.

* Clarify discontinuity comments

* Signal a stream discontinuity on restart due to stream error

* Apply suggestions from code review

Co-authored-by: uvjustin <46082645+uvjustin@users.noreply.github.com>

* Simplify stream discontinuity logic

frenck committed:
Add selectors to Scene service definitions (#46729)

jcam committed:
Fix missing color switch specific device class for Z-Wave JS driver >6.3 (#46718)

cgtobi committed:
Add selectors to Netatmo services (#46574)

* Add selectors

* Fix schedul selector

* Update homeassistant/components/netatmo/services.yaml

Co-authored-by: Bram Kragten <mail@bramkragten.nl>

* Update homeassistant/components/netatmo/services.yaml

Co-authored-by: Bram Kragten <mail@bramkragten.nl>

* Update services.yaml

* Added required field

Co-authored-by: Bram Kragten <mail@bramkragten.nl>

frenck committed:
Add selectors to Vacuum service definitions (#46728)

frenck committed:
Add selectors to Script service definitions (#46730)

frenck committed:
Add advanced selectors to Light service definitions (#46732)

frenck committed:
Add selectors to WLED service definitions (#46731)

frenck committed:
Add advanced selectors to Climate service definitions (#46736)

frenck committed:
Add selectors to Z-Wave JS service definitions (#46737)

frenck committed:
Add selectors to Logger, System Log & Logbook service definitions (#46740)

frenck committed:
Add selectors to MQTT service definitions (#46738)

frenck committed:
Add selectors to Color Extractor service definitions (#46742)

frenck committed:
Add selectors to Lock service definitions (#46743)

frenck committed:
Add selectors to Timer service definitions (#46744)

uvjustin committed:
Skip repeated segment in stream recorder (#46701)

* Skip repeated segment in stream recorder

* Allow for multiple overlap

spacegaier committed:
Fix typo in Tesla Powerwall strings (#46752)

frenck committed:
Add selectors to Hue service definitions (#46747)

frenck committed:
Add selectors to Media Player service definitions (#46739)

frenck committed:
Add selectors to Toon service definitions (#46750)

frenck committed:
Add selectors to Browser, Recorder, Shopping List service definitions (#46749)

frenck committed:
Add selectors to Conversation, Image Processing and Number service definitions (#46746)

frenck committed:
Minor service definition tweaks (#46741)

AdmiralStipe committed:
Added Slovenian language (sl-si) to Microsoft TTS (#46720)

frenck committed:
Add selectors to HomeKit service definitions (#46745)

frenck committed:
Add selectors to Twente Milieu service definitions (#46748)

frenck committed:
Ensure pre-commit's hassfest triggers on service file changes (#46753)

frenck committed:
Add selectors to Home Assistant service definitions (#46733)

Co-authored-by: Bram Kragten <mail@bramkragten.nl>

frenck committed:
Upgrade sentry-sdk to 0.20.3 (#46754)

emichael committed:
Allow LIFX bulbs to fade color even when off (#46596)

LIFX bulbs have the capability to fade their color attributes even while the bulb is off. When the bulb is later turned on, the fade will continue as if the bulb was on all along.

esev committed:
Add @esev as codeowner for wemo (#46756)

frenck committed:
Upgrade cryptography to 3.3.2 (#46759)

homeassistant committed:
[ci skip] Translation update

mdz committed:
Address late smarttub review (#46703)

* _config -> config

* remove unused string

* remove entity tests

* replace unit tests with integration tests using the core

* refactor polling to use asyncio.gather

* remove redundant component init

* remove gather in favor of simple loop

* use async_fire_time_changed instead of async_update_entity

* use hass.config_entries.async_setup instead of calling smarttub.async_setup_entry directly

* replace stray smarttub.async_setup_entry call

* async_unload_entry -> hass.config_entries.async_unload

* remove broken test

shbatm committed:
Fix flip-flopped substitutions in Custom Version Type Warning message. (#46768)

bdraco committed:
Implement percentage step sizes for fans (#46512)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

bdraco committed:
Fix backwards compatibility with vesync fans (#45950)

dependabot[bot] committed:
Bump actions/stale from v3.0.16 to v3.0.17 (#46777)

amelchio committed:
Raise ConditionError for zone errors (#46253)

* Raise ConditionError for zone errors

* Do not test missing state

* Handle multiple entities/zones

amelchio committed:
Raise ConditionError for and/or/not errors (#46767)

cgtobi committed:
Bump RMVtransport to 0.3.1 (#46780)

frenck committed:
Add Home Assistant color (#46751)

ludeeus committed:
Use string instead of slug for add-on slug validation (#46784)

oetken committed:
Allow multiple recipients for XMPP (#45328)

* recipient is a string list instead of a single string now
* this does NOT break existing automations/etc using this component

Commit:
Update GitHub Issue Form template (#46791)

This PR fixes the bug report template!

Details: https://gh-community.github.io/issue-template-feedback/changes/#what-do-i-need-to-do

raman325 committed:
Add zwave_js thermostat fan mode and fan state support (#46793)

* add thermostat fan mode and fan state support

* return when fan mode is not supported

* use get just in case

* validate state key is in states so we dont have to use get

* pylint

raman325 committed:
Format zwave_js dump as json (#46792)

homeassistant committed:
[ci skip] Translation update

esev committed:
Bump pywemo to 0.6.2 (#46797)

bdraco committed:
Implement suggested_area in the device registry (#45940)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

bdraco committed:
Cleanup inconsistencies in zha fan and make it a bit more dry (#46714)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

bdraco committed:
Add support for using a single endpoint for rest data (#46711)

emontnemery committed:
Add device_entities template function/filter (#46406)

bdraco committed:
Add suggested area support to nuheat (#46801)

bdraco committed:
Fix typing of fan speed count and steps (#46790)

bdraco committed:
Implement suggested area for netatmo (#46802)

bdraco committed:
Add suggested area to nexia (#46776)

bdraco committed:
Add suggested area to hunterdouglas_powerview (#46774)

bdraco committed:
Ensure recorder shuts down cleanly on restart before startup is finished (#46604)

bdraco committed:
Ensure HomeAssistant can still restart when a library file is missing (#46664)

bdraco committed:
Implement suggested area in lutron_caseta (#45941)

bdraco committed:
Implement percentage step sizes in HomeKit (#46722)

bdraco committed:
Update smarty to use new fan entity model (#45879)

bdraco committed:
Update alexa for new fan model (#45836)

bdraco committed:
Implement suggested areas in bond (#45942)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

bdraco committed:
Recover and restart the recorder if the sqlite database encounters corruption while running (#46612)

bdraco committed:
Add suggested area support to Sonos (#46794)

bdraco committed:
Run homekit service calls in async since the server is now async (#45859)

* Simplify homekit runs and service calls

Now that the homekit server is async, call_service
and run are running in the Home Assistant event loop

* remove comment

* remove another comment

amelchio committed:
Do not trigger when numeric_state is true at startup (#46424)

raman325 committed:
Bump zwave-js-server-python to 0.18.0 (#46787)

* updates to support changes in zwave-js-server-python

* bump lib version

* use named arguments for optional args

* re-add lost commits

allenporter committed:
Rollback stream StreamOutput refactoring in PR#46610 (#46684)

* Rollback PR#46610

* Update stream tests post-merge

fabaff committed:
Upgrade TwitterAPI to 2.6.6 (#46812)

ctalkington committed:
Update rokuecp to 0.8.0 (#46799)

* update rokuecp to 0.8.0

* Update requirements_test_all.txt

* Update requirements_all.txt

bdraco committed:
Add support for bond up and down lights (#46233)

uvjustin committed:
Add rtsp transport options to generic camera (#46623)

* Add rtsp transport options to generic camera

mib1185 committed:
change log level to info (#46823)

farmio committed:
Update xknx to 0.17.0 (#46809)

kpine committed:
Detect iBlinds v2.0 switch value as a cover not light (#46807)

* Remove unused "fibaro_fgs222" discovery hint

* Simplify multilevel switch current value discovery schema

* Force iBlinds v2.0 devices to be discovered as cover entities

* Rename discovery test file

ctalkington committed:
Implement suggested area in roku (#46819)

* implement suggested area in roku

* Update const.py

* Update test_media_player.py

* Update test_media_player.py

* Update test_media_player.py

* Update test_media_player.py

amelchio committed:
Explicitly create_task for asyncio.wait (#46325)

marecabo committed:
Validate icon and device_class of ESPHome sensor entities (#46709)

farmio committed:
Enable KNX routing optional local_ip (#46133)

farmio committed:
knx-read-service (#46670)

crazyfx1 committed:
Add open/close tilt support to KNX cover (#46583)

esev committed:
Bump pywemo to 0.6.3 (#46825)

esev committed:
Disable update polling for Wemo when devices can push updates (#46806)

MartinHjelmare committed:
Fix habitica entry unload clean up (#46798)

* Fix habitica entry unload clean up

* Fix service remove

* Add entry setup and unload test

* Fix config flow tests

* Assert service

esev committed:
Speed-up wemo discovery (#46821)

* Speed-up wemo discovery

* Use gather_with_concurrency to limit concurrent executor usage

* Comment fixup: asyncio executors -> executor threads

milanmeu committed:
Add Rituals Perfume Genie integration (#46218)

Co-authored-by: J. Nick Koston <nick@koston.org>

puddly committed:
Bump zigpy-znp from 0.3.0 to 0.4.0 (#46828)

mdz committed:
Add smarttub sensor platform and state sensor (#46775)

teharris1 committed:
Fix Insteon config flow with add X10 and device override (#45854)

homeassistant committed:
[ci skip] Translation update

StevenLooman committed:
Allow upnp ignore SSDP-discoveries (#46592)

G-Two committed:
Add new Subaru integration (#35760)

Co-authored-by: On Freund <onfreund@gmail.com>
Co-authored-by: J. Nick Koston <nick@koston.org>

emontnemery committed:
Add support for "alias" in script steps device, device_condition, and conditions (#46647)

Co-authored-by: Donnie <donniekarnsinsb@hotmail.com>

emontnemery committed:
Add support for disabling config entries (#46779)

bdraco committed:
Add support for preset modes in homekit fans (#45962)

ReneNulschDE committed:
Merge branch 'fix-OpenWeather---45517' of https://github.com/ReneNulschDE/core into fix-OpenWeather---45517

ReneNulschDE committed:
Add tests for default options
